client/state/current-user/gravatar-status/actions.js

```diff
@@ -34,20 +41,40 @@ export function uploadGravatar( file, bearerToken, email ) {
 					dispatch( {
 						type: GRAVATAR_UPLOAD_REQUEST_SUCCESS
 					} );
+					dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_success' ) );
 				} );
 				fileReader.readAsDataURL( file );
 			} )
 			.catch( () => {
 				dispatch( {
 					type: GRAVATAR_UPLOAD_REQUEST_FAILURE
 				} );
+				dispatch( recordTracksEvent( 'calypso_edit_gravatar_upload_failure' ) );
+				dispatch( bumpStat( 'calypso_gravatar_update_error', 'unsuccessful_http_response' ) );
```

Comment:
@dmsnell and I talked about this, and since using the analytics middleware is a big undertaking, I kept the analytics actions in the same place. I updated them to use `withAnalyitcs` in b077952b34447c5e2cc57d7b3094bb09c02819c6